icefoganalytics/travel-authorization
# Fixup: Unbreak Travel Desk Again

# Context

Travel desk /travel-desk page is broken again. Unbreak it.

# Implementation

Standardize TravelDeskQuestion model.
Unbreak legacy travel desk save function.

# Screenshots

![image](https://github.com/user-attachments/assets/926c2a8f-de12-4e7e-83d7-15fd2db3c9fb)
![image](https://github.com/user-attachments/assets/d9105f59-9b16-4c4d-b72b-57a27be67bf9)

Options Provided
![image](https://github.com/user-attachments/assets/63b3ef93-c4f6-45f6-960e-75fc49140f0c)

# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
2. Boot the app via `dev up`
3. Log in to the app at http://localhost:8080
4. From the top right kebab menu, select administration.
5. Click on "User Managment" on the left.
6. Find your user record and click on it.
7. Add the "td_user" role to your user and click save.
8. From the top drop down nav, select Travel Desk.
9. If there aren't any records, go to the My Travel Requests page and complete process until you submit a request, and then come back here.
10. Click on the "pencil" icon to the right of the request to edit the request.
11. Check that the modal opens without crashing, and that you can "Save Draft".
12. Check that you can "Send to Traveler".
13. Go to the "My Travel Request" page and click on the request you just send back.
14. Go to the "Request" tab, and check that the status shows as "options_provided"
